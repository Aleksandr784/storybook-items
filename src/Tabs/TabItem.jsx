import React from 'react';
import './TabItem.css';

export function TabItem ({text}) {
    const [showMessage, setShowMessage] = React.useState("button not clicked")
 const clickMe = () => {
        console.log("ki")
        setShowMessage ("button clicked") 
        
    }
    
    return (
        <div className="tab-item">
<button className="button" onClick={clickMe} >{text}</button>
{showMessage}
        </div>
    )
}