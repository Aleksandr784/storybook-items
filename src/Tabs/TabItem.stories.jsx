import React from "react";
import { TabItem } from "./TabItem";

export default {
    title: 'TabItem',
    component: TabItem,
    argTypes: {
        text: {control: 'text'}
    }
}

const Template = (args) => <TabItem {...args} />

export const Base = Template.bind({})
Base.args = {
    text: "initial text"
}