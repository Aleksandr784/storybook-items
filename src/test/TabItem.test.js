import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import {TabItem} from "../Tabs/TabItem";
import React from 'react';



test('Text change after button click', () => {
    render(<TabItem text="Button click" />);
    const buttonElement = screen.getByText(/Button click/i);
    expect(buttonElement).toBeInTheDocument();
    const text=screen.getByText("button not clicked");
    expect(text).toBeInTheDocument();

 
 
  userEvent.click(screen.getByText('Button click'))
  const textChanged = screen.getByText("button clicked")
  expect(textChanged).toBeInTheDocument();

  
});