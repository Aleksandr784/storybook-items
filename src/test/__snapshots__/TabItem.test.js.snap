// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text change after button click 1`] = `
<div
  className="tab-item"
>
  <button
    className="button"
    onClick={[Function]}
  >
    Button click
  </button>
  button not clicked
</div>
`;
